File: cpe49題/10101 Bangla Numbers.cpp
#include <bits/stdc++.h>

using namespace std;

void bangla(long long n){
	int kuti,lakh,hajar,shata;
	if(n>=10000000){
		kuti=n/10000000;		
		bangla(kuti);
		n%=10000000;
		cout<<" kuti";
	}
	if(n>=100000){
		lakh=n/100000;
		n%=100000;
		cout<<" "<<lakh<<" lakh";
	}
	if(n>=1000){
		hajar=n/1000;
		n%=1000;
		cout<<" "<<hajar<<" hajar";
	}
	if(n>=100){
		shata=n/100;
		n%=100;
		cout<<" "<<shata<<" shata";	
	}
	if(n){
		cout<<" "<<n;
	}
}

int main(){
	long long n;
	int caseNum=0;
	while(cin>>n){
		cout<<setw(4)<<right<<++caseNum<<".";
		if(n){
			bangla(n);
		}else{
			cout<<" 0";
		}
		
		cout<<endl;
	}
}

File: cpe49題/10101 Bangla Numbers.py
def bangla(num):
	if num>=10000000:	
		bangla(num//10000000)
		print(f" kuti",end="")
		num%=10000000
	if num>=100000:
		bangla(num//100000)
		print(f" lakh",end="")	
		num%=100000
	if num>=1000:
		bangla(num//1000)
		print(f" hajar",end="")	
		num%=1000
	if num>=100:
		bangla(num//100)
		print(f" shata",end="")
		num%=100
	if num>0:
		print(f" {num}", end="")

caseNum=1
while True:
	try:
		n=int(input())
	except EOFError:
		break
	
	if n==0:
		print(f"{caseNum:4}. 0",end="")
	else:
		print(f"{caseNum:4}.",end="")
		bangla(n)
	print()
	caseNum+=1
